### PR TITLE
Feature/26s - Expose webpack-plugin-serve

### LINKED ISSUE #26: Add webpack-plugin-serve setup
---

PR #27

---

**About**

Since we're using [webpack-plugin-serve](https://www.npmjs.com/package/webpack-plugin-serve) based setup in the projects now, it makes sense to provide here as a function.

**Possible implementation**

See https://github.com/pmedianetwork/adverity-presense-frontend/pull/839 and adapt so that it works with the other two projects.

---

**Analysis**
_Should be filled out in the Analysis phase. Needs to be approved by another engineer before implementation starts. See [How to write a good analysis](http://handbook.adverity.net/kanban/#how-to-write-an-analysis-of-an-issue)_

---

**Tasks**
_A very detailed and technical list of tasks. Should be filled out after Analysis._

- [ ] task 1
- [ ] ..

### PR DESCRIPTION
**Pull request by bebraw**
---
Closes #26

---

**How to test (feature)**

I could add it to the setup. Now the setup is simply a highly customizable copy that should work across projects.

**How to test (potential regressions)**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -698,6 +745,7 @@ export {
   extractCSS,
   dontParse,
   webpackDevServer,
+  webpackPluginServe,
```
Comment (TkDodo) [inline src/index.ts:693]: just a quick thought:

- is any product still using `webpack-dev-server` ? If not, we can probably remove it for consistency.
- the same for `reactHotLoader` - the viewer options we provide the better

**Review by TkDodo — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (bebraw) [reply src/index.ts:693]: Yeah, that's what I was thinking too. Technically we could drop `webpackDevServer` and `reactHotLoader` and make this a major version.

Comment (TkDodo) [reply src/index.ts:693]: sounds good!

Comment (bebraw) [reply src/index.ts:693]: Ok, done. As a side effect now we have less dependencies to download.

File: src/index.ts [hunk 2]
```diff
@@ -397,57 +400,46 @@ function dontParse(paths: webpack.Module["noParse"]): webpack.Configuration {
...
-    },
-    plugins: [new webpack.HotModuleReplacementPlugin()],
-  };
-}
-
-// This is the legacy option for React projects. It requires you to use
-// hot wrapper from react-hot-loader at the root of an app.
-function reactHotLoader(): webpack.Configuration {
+// Note that when using webpack-plugin-serve, you have to run
+// the process through regular webpack, not webpack-dev-server!
```
Comment (TkDodo) [inline src/index.ts]: is this comment still up to date, given that we dropped `webpack-dev-server` now?

**Review by TkDodo — CHANGES_REQUESTED** [review]

**Comment by bebraw** [comment]
Potentially yes but I can remove it. The thing is that it will run through webpack-dev-server if you run it like this by accident but it’s not going to give the intended result.

File: src/index.ts (hunk 2 above)
Comment (bebraw) [reply src/index.ts]: I improved the wording.

**Review by TkDodo — APPROVED** [review]

**Pull request merged**